### PR TITLE
[WIP][No-28][training]show-list-blog[REVIEW]

### PR DESCRIPTION
**Pull request by phamhieulftv**
Create table : 
- php artisan migrate
Create data : 
- php artisan db:seed
Run project : 
- php artisan  serve
Login to see all blog with account : 
user@lifull-tech.vn/lifull@123

### CODE REVIEW

File: app/Blog.php [hunk 1]
```diff
@@ -9,4 +9,13 @@ class Blog extends Model
     protected $table = 'blogs';
     const IS_FALSE = 0;
     const IS_TRUE = 1;
+
```
Comment (loitq) [inline app/Blog.php]: add comment for function

File: database/seeds/BlogsTableSeeder.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+class BlogsTableSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        factory(App\Blog::class, 50)->create();
```
Comment (loitq) [inline database/seeds/BlogsTableSeeder.php]: use define for magic number.

File: public/asset/admin/js/blogs.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+$(function() {
+    $("a").click(function() {
+        alert("on click");
+    })
+});
```
Comment (loitq) [inline public/asset/admin/js/blogs.js]: enter end of line by stand psr2

File: routes/web.php [hunk 4]
```diff
@@ -47,6 +47,8 @@
     //Route User
     Route::namespace('User')->group(function () {
         Route::get('/', 'UserController@index')->name('user.index');
+        Route::post('/comment', 'UserController@getComment')->name('user.comment');
+        Route::post('/add-comment', 'UserController@addComment')->name('user.addComment');
```
Comment (loitq) [inline routes/web.php]: use laravel resful for resoure when create name function.

**Review by loitq — COMMENTED** [review]
Please check.

File: app/Blog.php (hunk 1 above)
Comment (phamhieulftv) [reply app/Blog.php]: i fixed

File: database/seeds/BlogsTableSeeder.php (hunk 2 above)
Comment (phamhieulftv) [reply database/seeds/BlogsTableSeeder.php]: i fixed

File: public/asset/admin/js/blogs.js (hunk 3 above)
Comment (phamhieulftv) [reply public/asset/admin/js/blogs.js]: i fixed

File: routes/web.php (hunk 4 above)
Comment (phamhieulftv) [reply routes/web.php]: i fixed

Comment (loitq) [reply routes/web.php]: can you fix...
```
Route::get('blog/comment', 'CommentsController@index')->name('blog.comment');
Route::post('blog/{id}/comment', 'CommentsController@store')->name('blog.comment.store');
```

File: routes/web.php [hunk 5]
```diff
@@ -47,6 +47,8 @@
     //Route User
     Route::namespace('User')->group(function () {
         Route::get('/', 'UserController@index')->name('user.index');
+        Route::get('/blog/{id}/comment', 'CommentController@index')->name('blog.comment.index');
+        Route::post('/blog/comment/create', 'CommentController@create')->name('blog.comment.create');
```
Comment (loitq) [inline routes/web.php]:
```
Route::post('/blog/{id}/comment', 'CommentController@create')->name('blog.comment.store');
```

File: app/Http/Controllers/User/CommentController.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            'blogId' => $request['blogId'],
+            'listComments' => $listComments
+        ];
+    }
+
+    /**
+     * Handle get list comment by blogId.
+     *
+     * @param Request $request
+     *
```
Comment (loitq) [inline app/Http/Controllers/User/CommentController.php]: remove line.

File: app/Http/Controllers/User/CommentController.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    public function index(Request $request, $blogId)
+    {
+        return $listComments = $this->getCommentsbyBlogId($blogId);
+    }
+
+    /**
+     * Handle add new comment and load new list comment.
+     *
+     * @param Request $request
+     *
```
Comment (loitq) [inline app/Http/Controllers/User/CommentController.php]: remove line

**Review by loitq — COMMENTED** [review]
I comment... please fix

File: routes/web.php (hunk 5 above)
Comment (phamhieulftv) [reply routes/web.php]: i fixed

File: app/Http/Controllers/User/CommentController.php (hunk 6 above)
Comment (phamhieulftv) [reply app/Http/Controllers/User/CommentController.php]: i fixed

File: app/Http/Controllers/User/CommentController.php (hunk 7 above)
Comment (phamhieulftv) [reply app/Http/Controllers/User/CommentController.php]: i fixed

**Review by loitq — APPROVED** [review]
LGTM.

File: app/Http/Controllers/User/CommentController.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    public function create(Request $request)
+    {
+        $user = Auth::user();
+        $comment =  new Comments();
+
+        $comment->user_id = $user->id;
+        $comment->blog_id = $request['blogId'];
+        $comment->comment_content = $request->comment_content;
+
+        $comment->save();
```
Comment (ghost) [inline app/Http/Controllers/User/CommentController.php]: Not sure if save () is successful
=> So need to check more error case (Try .. catch)

- Add return message error

Comment (phamhieulftv) [reply app/Http/Controllers/User/CommentController.php]: I fixed

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
LGTM

**Pull request closed without merging**